### DISCUSSION TITLE
Test dataset feature engineering 에서 train dataset 의 통계량 사용하기

### DISCUSSION THREAD

**Discussion by ddoyoon** [bcaitech1/p4-dkt-lastffang#118] [Features]
지금 저희 코드에서는 train, test data 에 각각 feature engineering 을 해주고 있는데
현재 마스터 브랜치에서 사용하고 있는 feature 중 지식 태그 분류별 정답률, 시험지 별 정답률, 문항별 정답률 등 유저에 따라 달라지지 않는 값들은 train 에서 통계량을 구하고 그 값을 test 에서도 사용하는게 어떨까요?

각 데이터에서 따로 구한 통계량이 얼마나 차이가 나는지는 확인을 못 해봤지만, test data 안에서만 통계량을 구하기에는 test set 자체가 너무 작은 것 같습니다. 
참고하고 있는 서일님 feature engineering 코드에서도 태그/시험지/문항별 정답률의 평균과 총 정답 개수는 train dataset 에서 미리 구한 값으로 train 과 test 양쪽에서 사용하고 있는 것 같습니다..!

**Comment by GY-Jeong** [comment]
좋은 의견이라는 생각이 듭니다!
만약에 train에서 특정 knowledgeTag에 대한 문제가 10개, 정답률이 0.5가 나와서 그 값으로 학습이 되었는데,
test에서 특정 knowledgeTag에 대한 문제가 1개, 정답률이 1.0이 나오면 너무 편향된 정보로 inference 되는 문제가 있네요

**Reply by GY-Jeong** [reply]
이슈로 보내버립시다,,

**Comment by scarlettJin** [comment]
오! 우리도 그렇게 가야 겠네요!!!
